playground/frontend/playground_components/test/src/repositories/emulator_type_grpc_extension_test.dart

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:playground_components/src/api/v1/api.pbgrpc.dart' as g;
+import 'package:playground_components/src/repositories/emulator_type_grpc_extension.dart';
+
+void main() {
+  group('Emulator type extensions test', () {
+    for (final value in g.EmulatorType.values) {
+      test('Emulator type ${value.name} converts to the same value', () {
+        expect(
+          value.model?.grpc ?? g.EmulatorType.EMULATOR_TYPE_UNSPECIFIED,
```

Comment:
If any of `g.EmulatorType.values`'s `model.grpc` returns null then it's an error we better know about.

Reply:
Oh, it's the grpc's model, it is allowed to return null, never mind.